packages/react-instantsearch-core/src/connectors/__tests__/connectDynamicWidgets.js

```diff
@@ -0,0 +1,284 @@
+import { SearchResults, SearchParameters } from 'algoliasearch-helper';
+import connector from '../connectDynamicWidgets';
+
+jest.mock('../../core/createConnector', () => x => x);
+
+const EMPTY_RESPONSE = {
+  results: [
+    {
+      hits: [],
+      nbHits: 0,
+      page: 0,
+      nbPages: 0,
+      hitsPerPage: 20,
+      exhaustiveNbHits: true,
+      query: '',
+      queryAfterRemoval: '',
+      params:
+        'highlightPreTag=%3Cais-highlight-0000000000%3E&highlightPostTag=%3C%2Fais-highlight-0000000000%3E&query=&facets=%5B%5D&tagFilters=',
+      index: 'instant_search',
+      processingTimeMS: 2,
+    },
+  ],
+};
+
+describe('connectDynamicWidgets', () => {
+  const empty = {};
+
+  describe('single index', () => {
+    const contextValue = {
+      mainTargetedIndex: 'index',
+    };
+
+    const createSingleIndexSearchResults = (result = {}, state) => ({
+      results: new SearchResults(new SearchParameters(state), [
+        {
+          ...EMPTY_RESPONSE.results[0],
+          ...result,
+        },
+      ]),
+    });
+
+    describe('getProvidedProps', () => {
+      it('empty results', () => {
+        const props = { contextValue };
+        const searchState = {};
+        const searchResults = empty;
+
+        const actual = connector.getProvidedProps(
+          props,
+          searchState,
+          searchResults
+        );
+
+        const expectation = {
+          attributesToRender: [],
+        };
+
+        expect(actual).toEqual(expectation);
+      });
+
+      it('error without transformItems', () => {
+        const props = { contextValue };
+        const searchState = {};
+        const searchResults = createSingleIndexSearchResults();
+
+        expect(() => {
+          connector.getProvidedProps(props, searchState, searchResults);
+        }).toThrowErrorMatchingInlineSnapshot(
+          `"props.transformItems is not a function"`
```

Comment:
Until we get data from the engine, we'd better provide a concrete error message for missing `transformItems`.